enhancements/microshift/microshift-router-configuration.md

```diff
@@ -0,0 +1,515 @@
+  nodeIP: 192.168.122.254
+$ oc get svc -n openshift-ingress router-default -o json | jq '.spec.clusterIP'
+"10.43.252.173"
+$ oc get svc -n openshift-ingress router-default -o json | jq '.status'
+{
+  "loadBalancer": {
+    "ingress": [
+      {
+        "ip": "192.168.122.254"
+      }
+    ]
+  }
+}
+$ oc get svc -n openshift-ingress router-default -o json | jq '.status.loadBalancer.ingress[]'
+{
+  "ip": "192.168.122.254"
+}
+$ sudo iptables -t nat -S
+...
+-A OVN-KUBE-EXTERNALIP -d 192.168.122.254/32 -p tcp -m tcp --dport 443 -j DNAT --to-destination 10.43.252.173:443
+-A OVN-KUBE-EXTERNALIP -d 192.168.122.254/32 -p tcp -m tcp --dport 80 -j DNAT --to-destination 10.43.252.173:80
+...
+```
+
+#### Firewalling ports
+Using LoadBalancer service will create special iptables rules with more
+precedence than those from firewalld. This means a `LoadBalancer` service is
+immune to such configurations. In order to firewall ports the user needs to
+take action by either creating `NetworkPolicy` resources or disabling the
+router.
+
+Disabling the router might be disruptive, as not all traffic may come from
+external sources. This procedure also requires a restart, which might not be
+desirable/possible in all situations.
+
+Using `NetworkPolicy` a user may control the traffic that is allowed into
+the cluster. By creating a deny-all policy the router will still remain
+operative for internal traffic, while rejecting all external traffic. An
+example follows:
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: block-router
+  namespace: openshift-ingress
+spec:
+  ingress:
+  - from:
+    - ipBlock:
+        cidr: 10.42.0.0/16
+  podSelector:
+    matchLabels:
+      ingresscontroller.operator.openshift.io/deployment-ingresscontroller: default
+  policyTypes:
+  - Ingress
+```
+Assuming the pod network is configured to use CIDR `10.42.0.0/16`, the above
+policy will block all traffic not coming from that network.
+
+Any of these methods is equivalent to using firewalld to block traffic, but
+using the CNI instead of a host level setting. This provides greater
+flexibility, as it is self contained within MicroShift APIs.
+
+If a user needs to block traffic to a port this needs a `NetworkPolicy`
+resource like the one shown above. This is NOT automatically done by
+MicroShift, user action is required.
+
+#### Exposing the router
+Relying on [LoadBalancer controller enhancement](loadbalancer-service-support.md)
+means the services are updated by this component. However, ovnk only turns IP
+addresses into iptables rules, it does not take Hostname, which is another
+valid field for the service status.
+
+Since the configuration uses NIC names and also host names, the controller
+needs to translate these to their corresponding IP addresses, and then
+configure them in the service. To keep up with the dynamic nature of IP
+addresses and names, it needs to perform periodic checks and updates.
```

Comment:
Does this name resolution have well defined semantics?  

* Where is the resolution done?  For example, is it done on my machine from which I install MicroShift, or from within MicroShift?
* Which name server is used?  For example, does it use the cluster DNS service, or whatever is in the MicroShift node host's `/etc/resolv.conf`, or something else?  Could I use a host name that I specified in `/etc/hosts`?  
* How often is name resolution done?
* If I specify a host name that can resolve when I install MicroShift but then the host name cannot be resolved after installation (maybe because the record was deleted, or maybe because DNS is down), what happens?

Reply:
Oh, one more question:  Are IPv6 addresses included in the name resolution and service status?

Reply:
We dropped the DNS resolution so this now only applies to NICs. We do not have IPv6 support yet, as that is a feature for a future version.